reactivesocket-client/src/main/java/io/reactivesocket/client/LoadBalancer.java

```diff
@@ -151,22 +250,29 @@ private synchronized void refreshAperture() {
 
         long now = Clock.now();
         boolean underRateLimit = now - lastApertureRefresh > APERTURE_REFRESH_PERIOD;
-        int previous = targetAperture;
         if (avgPending < 1.0 && underRateLimit) {
-            targetAperture--;
-            lastApertureRefresh = now;
-            pendings.reset((MIN_PENDINGS + MAX_PENDINGS)/2);
+            updateAperture(targetAperture - 1, now);
         } else if (2.0 < avgPending && underRateLimit) {
-            targetAperture++;
-            lastApertureRefresh = now;
-            pendings.reset((MIN_PENDINGS + MAX_PENDINGS)/2);
+            updateAperture(targetAperture + 1, now);
         }
-        targetAperture = Math.max(MIN_APERTURE, targetAperture);
-        int maxAperture = Math.min(MAX_APERTURE, activeFactories.size());
+    }
+
+    /**
+     * Update the aperture value and ensure its value stays in the right range.
+     * @param newValue new aperture value
+     * @param now time of the change (for rate limiting purposes)
+     */
+    private void updateAperture(int newValue, long now) {
+        int previous = targetAperture;
+        targetAperture = newValue;
+        targetAperture = Math.max(minAperture, targetAperture);
+        int maxAperture = Math.min(this.maxAperture, activeSockets.size() + activeFactories.size());
         targetAperture = Math.min(maxAperture, targetAperture);
+        lastApertureRefresh = now;
+        pendings.reset((minPendings + maxPendings)/2);
 
         if (targetAperture != previous) {
-            logger.info("Current pending=" + avgPending
+            logger.info("Current pending=" + pendings.value()
```

Comment:
They could, in practice it's not that chatty in steady-state.
I'll update the code.